Otherworldly Pines/Assets/Scripts/MovingPlatform.cs

```diff
@@ -90,6 +78,16 @@ void OnCollisionEnter2D(Collision2D target)
         if (!isVertical && !isPaused) {
             transform.position += (Vector3) (GetCurrentDirection() * Time.deltaTime * movementSpeed);
         }
+
+        if (Physics2D.OverlapArea(new Vector2(transform.position.x - 0.7f, transform.position.y -0.3f),
```

Comment:
I'm pretty sure you can use `collider.bounds.min` and `collider.bounds.max` to get the top left and bottom right corners of the collider.